src/config/loader.ts

```diff
@@ -110,6 +112,19 @@ async function _loadUserConfig(
           configs.rc?.[key] ??
           configs.packageJson?.[key]) as NitroConfig[K];
 
+      const configRootDir =
+        configs.main?.rootDir ??
+        configs.rc?.rootDir ??
+        configs.packageJson?.rootDir;
+      if (
+        configRootDir &&
+        resolve(configRootDir) !== resolve(configOverrides.rootDir || ".")
+      ) {
+        consola.warn(
+          `Nitro config file should be in the \`rootDir: '${configRootDir}'\`.`
```

Comment:
The warning is a little misleading. Simply, we are warning that the `rootDir` specified in the config has been overridden by loader (Nitro CLI or other framework)

But honestly, I'm not even sure if it is best idea or if makes more false positive warnings.. We could also make `defineNitroConfig` type better to remove `rootDir` (as breaking change)

Reply:
Removing `rootDir` from `defineNitroConfig` would be better if it's not meant to be used in the config file, IMO. Less work and also be less confusing.